public/app/features/explore/QueryLibrary/QueryLibraryContext.tsx

```diff
@@ -0,0 +1,89 @@
+import { PropsWithChildren, useState, createContext, useContext, useCallback } from 'react';
+
+import { config } from '@grafana/runtime';
+import { DataQuery } from '@grafana/schema/dist/esm/veneer/common.types';
+
+/**
+ * Context with state and action to interact with Query Library. The Query Library feature consists of a drawer
+ * that shows existing queries and allows users to use them and manage them and then an AddQueryModal which allows
+ * users to save a query into the library. Both of those are included in Grafana AppChrome component.
+ *
+ * Use this context to interact with those components, showing, hiding and setting initial state for them.
+ */
+type QueryLibraryContextType = {
+  queryLibraryAvailable: boolean;
+  drawerOpened: boolean;
+
+  activeDatasources: string[] | undefined;
+  openDrawer: (datasources: string[] | undefined) => void;
+  closeDrawer: () => void;
+
+  activeQuery: DataQuery | undefined;
+  addQueryModalOpened: boolean;
+  openAddQueryModal: (query: DataQuery) => void;
+  closeAddQueryModal: () => void;
+};
+
+export const QueryLibraryContext = createContext<QueryLibraryContextType>({
+  queryLibraryAvailable: false,
+  drawerOpened: false,
+
+  activeDatasources: undefined,
+  openDrawer: () => {},
+  closeDrawer: () => {},
+
+  activeQuery: undefined,
+  addQueryModalOpened: false,
+  openAddQueryModal: () => {},
+  closeAddQueryModal: () => {},
+});
+
+export function useQueryLibraryContext() {
+  return useContext(QueryLibraryContext);
+}
+
+export function QueryLibraryDrawerContextProvider({ children }: PropsWithChildren) {
```

Comment:
nit: can we just call this `QueryLibraryContextProvider`? `Drawer` feels like an implementation detail